docs/readmes/README.featureflags.md

```diff
@@ -48,3 +48,11 @@ their use cases.
 
 Aad-pod-identity has a new flag `immutable-user-msis` which can be used to prevent deletion of specified identities from VM/VMSS.
 The list is comma separated. Example: 00000000-0000-0000-0000-000000000000,11111111-1111-1111-1111-111111111111
+
+## Metadata header required flag
+
+When you query the Instance Metadata Service, you must provide the header `Metadata: true` to ensure the request was not unintentionally redirected. Your can see examples of this header in [the Azure documentation](https://docs.microsoft.com/en-us/azure/virtual-machines/linux/instance-metadata-service#using-headers).
+
+This is critical especially when you [acquire an access token](https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/how-to-use-vm-token#get-a-token-using-http) as a mitigation againt Server Side Request Forgery (SSRF) attack. 
+
+The `metadataHeaderRequired` flag for NMI will block all requests without Metadata header and return an HTTP 400 response. This flag is disabled by default for compability, but recommended for usrs to enable this feature.
```

Comment:
can you also add the flag used to enable outside of helm? 

Something like `Add the following arg --metadata-header-required=true to NMI daemonset`

Reply:
Done, but set is to `false` for compability